lib/galaxy/datatypes/converters/gff_to_bed_converter.py

```diff
@@ -1,33 +1,118 @@
+
+    # Build lists for transcript blocks' starts, sizes.
+    block_starts, block_ends = zip(*blocks)
+    # Get transcript start, end.
+    t_start = min(block_starts)
+    t_end = max(block_ends)
+    # Calculate block sizes (as str).
+    block_sizes = [str(block_end - block_start) for block_start, block_end in blocks]
+    # Adjust block starts, convert to str
+    block_starts = [str(block_start - t_start) for block_start in block_starts]
+
+    # BED12 format: chrom, chromStart, chromEnd, name, score, strand,
+    # thickStart, thickEnd, itemRgb, blockCount, blockSizes, blockStarts
+    #
+    # Render complete feature with thick blocks. There's no clear way to do this unless
+    # we analyze the block names, but making everything thick makes more sense than
+    # making everything thin.
+    return "%s\t%i\t%i\t%s\t0\t%s\t%i\t%i\t0\t%i\t%s\t%s\n" % (
+        chrom,
+        t_start,
+        t_end,
+        name,
+        strand,
+        t_start,
+        t_end,
+        len(block_starts),
+        ",".join(block_sizes),
+        ",".join(block_starts)
+    )
 
 
 def __main__():
-    input_name = sys.argv[1]
-    output_name = sys.argv[2]
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--bedcols', type=int, choices=[6, 12], help="Type of BED to produce, either bed6 or bed12")
+    parser.add_argument('input', type=str)
+    parser.add_argument('output', type=str)
+    args = parser.parse_args()
+    input_name = args.input
+    output_name = args.output
+    bedcols = args.bedcols
+
     skipped_lines = 0
     first_skipped_line = 0
     i = 0
-    with open(input_name) as fh, open(output_name, 'w') as out:
-        for i, line in enumerate(fh):
+    if bedcols == 12:
+        from galaxy.datatypes.util.gff_util import parse_gff_attributes
+        cur_transcript_chrome = None
+        cur_transcript_id = None
+        cur_transcript_strand = None
+        cur_transcripts_blocks = []  # (start, end) for each block
+    with open(output_name, 'w') as out, open(input_name) as in_fh:
+        for i, line in enumerate(in_fh):
             line = line.rstrip('\r\n')
             if line and not line.startswith('#'):
                 try:
+                    # GFF format: chrom source, name, chromStart, chromEnd, score, strand, attributes
                     elems = line.split('\t')
                     start = str(int(elems[3]) - 1)
                     strand = elems[6]
                     if strand not in ['+', '-']:
                         strand = '+'
-                    # GFF format: chrom source, name, chromStart, chromEnd, score, strand
-                    # Bed format: chrom, chromStart, chromEnd, name, score, strand
-                    #
-                    # Replace any spaces in the name with underscores so UCSC will not complain
+                    # Replace any spaces in the name with underscores so UCSC will not complain.
                     name = elems[2].replace(" ", "_")
```

Comment:
This should be replaced with something like: `name = t_id or attributes.get('gene_id') or attributes.get('ID') or elems[2].replace(" ", "_")` , but we would need to also move lines 79 and 80 before this one and `parse_gff_attributes` would have to be imported also for the gff-to-bed6 converter.
The import is presently problematic because there is no conda package for [galaxy-data](https://pypi.org/project/galaxy-data/) yet, and also the PyPI version is quite old (19.9.0.dev3) and still a pre-release.

Reply:
So it sound to me like adding a recipe for galaxy-data is a good next step: https://github.com/bioconda/bioconda-recipes/pull/26365